src/test_application/main.cpp

```diff
@@ -309,13 +317,61 @@ void debug_print_help()
         << "* flight    - Create an ActiveArea and start Magnum\n"
         << "\n"
         << "Other things to type:\n"
+        << "* list_pkg  - List Packages and Resources\n"
         << "* list_uni  - List Satellites in the universe\n"
         << "* list_ent  - List Entities in active scene\n"
         << "* list_upd  - List Update order from active scene\n"
         << "* help      - Show this again\n"
         << "* exit      - Deallocate everything and return memory to OS\n";
 }
 
+template <typename RES_T>
+void debug_print_resource_group(osp::Package& rPkg)
+{
+    osp::Package::group_t<RES_T> const &group = rPkg.group_get<RES_T>();
+
+    if (group.empty())
+    {
+        return;
+    }
+
+    std::cout << "  * TYPE: " << entt::type_name<RES_T>().value() << "\n";
+
+    for (auto const& [key, resource] : group)
+    {
+        std::cout << "    * " << (resource.m_data.has_value() ? "LOADED" : "RESERVED") << ": " << key << "\n";
+    }
+}
+
+void debug_print_resources()
+{
+    std::vector<osp::Package*> packages = {
+        &g_osp.debug_find_package("lzdb")};
+
+    if (g_ospMagnum != nullptr)
+    {
+        packages.push_back(&g_ospMagnum->get_context_resources());
+    }
+
+    for (osp::Package* pPkg : packages)
+    {
+        osp::Package &rPkg = *pPkg;
+        std::cout << "* PACKAGE: " << rPkg.get_prefix() << "\n";
+
+        debug_print_resource_group<osp::PrototypePart>(rPkg);
+        debug_print_resource_group<osp::BlueprintVehicle>(rPkg);
+
+        debug_print_resource_group<Magnum::Trade::ImageData2D>(rPkg);
+        debug_print_resource_group<Magnum::Trade::MeshData>(rPkg);
+        debug_print_resource_group<Magnum::GL::Texture2D>(rPkg);
+        debug_print_resource_group<Magnum::GL::Mesh>(rPkg);
+
+        debug_print_resource_group<adera::active::machines::ShipResourceType>(rPkg);
+        debug_print_resource_group<adera::shader::PlumeShader>(rPkg);
+        debug_print_resource_group<adera::shader::Phong>(rPkg);
```

Comment:
I guess it's possible to add some polymorphic functionality to Package to list resource types, but this works for now.